pkg/cmd/cli/cmd/logs.go

```diff
@@ -189,5 +218,32 @@ func (o OpenShiftLogsOptions) RunLog() error {
 		// Use our own options object.
 		o.KubeLogOptions.Options = o.Options
 	}
-	return o.KubeLogOptions.RunLogs()
+	if !o.JenkinsLogs {
+		return o.KubeLogOptions.RunLogs()
+	}
+
+	bc, isBC := o.KubeLogOptions.Object.(*buildapi.BuildConfig)
+	build, isBld := o.KubeLogOptions.Object.(*buildapi.Build)
+	switch {
+	case isBC:
+		buildName := buildutil.BuildNameForConfigVersion(bc.ObjectMeta.Name, int(bc.Status.LastVersion))
+		build, _ = o.Client.Builds(o.Namespace).Get(buildName, metav1.GetOptions{})
+		if build == nil {
+			return errors.New(fmt.Sprintf("The build %s for build config %s was not found", buildName, bc.ObjectMeta.Name))
+		}
+		fallthrough
```

Comment:
(or possibly you can drop the isBC and isBld variables and just check for bc==nil and build==nil in the switch... i'm not sure if those end up nil when a cast fails or not, but i would assume they do.